### PR TITLE
LLM: Enable attempting loading method automatically

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

https://github.com/intel-analytics/BigDL/issues/8837#issuecomment-1696883932

We are enforcing the experimental feature "low_cpu_mem_usage" to alleviate memory pressure when loading large models. This feature avoids initializing the parameters when initializing empty model itself (by put everything on a meta device). However,  some models like the bark contains a operation named "weights_norm" , which normalizes the weights during initialization, necessitating non-empty weights. This conflicts with the design of the meta device, and this is why "low_cpu_mem_usage" doesn't work on bark.

So in this PR,  we first load model with "low_mem_cpu_usage=True", if an exception is caught, then throw a warning and proceed to handle it using the original loading method.
```log
(changmim-llm) cpx@cpx-2:/disk1/changmin/BigDL/python/llm$ python test_bark.py 
Failed to load models with `low_cpu_mem_usage` specified, will fall to traditional load method with higher memory consumption.

/disk1/changmin/BigDL/python/llm/src/bigdl/llm/transformers/convert.py:103: UserWarning: No linear modules were found in your model. This can happen for some architectures such as gpt2 that uses Conv1D instead of Linear layers. Please double check your model architecture, or submit an issue on github if you think this is a bug.
  warnings.warn(
Cost time--1:  0.009249687194824219
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:10000 for open-end generation.
Cost time--2:  36.95219945907593
Cost time:  36.9615421295166
save audio to bark_out.wav---
```

### CODE REVIEW

File: python/llm/src/bigdl/llm/transformers/convert.py [hunk 1]
```diff
@@ -85,12 +85,13 @@ def _replace_with_quant_linear(model, qtype, modules_to_not_convert=None,
...
         # Remove the last key for recursion
         if len(list(module.children())) > 0:
-            _, has_been_replaced = _replace_with_quant_linear(
+            _, _flag = _replace_with_quant_linear(
                 module,
                 qtype,
                 modules_to_not_convert,
                 current_key_name,
             )
+            has_been_replaced = _flag or has_been_replaced
```
Comment (leonardozcm) [inline python/llm/src/bigdl/llm/transformers/convert.py:94]: Otherwise has_been_replaced will always be overwritten by the last one iteration.

File: python/llm/src/bigdl/llm/transformers/model.py [hunk 2]
```diff
@@ -98,7 +102,19 @@ def load_convert(cls, q_k, optimize_model, *args, **kwargs):
...
-        model = cls.HF_Model.from_pretrained(*args, **kwargs)
+        # In case it needs a second try,
+        # `from_pretrained`` may pop items out in dict
+        # and lead to args missing.
+        _args = copy.deepcopy(args)
+        _kwargs = copy.deepcopy(kwargs)
+        try:
+            model = cls.HF_Model.from_pretrained(*args, **kwargs)
+        except NotImplementedError:
+            logger.warning("Failed to load models with `low_cpu_mem_usage` specified, "
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/model.py:113]: info instead of warning
